src/util/triplea/MapXMLCreator/MapXMLHelper.java

```diff
@@ -0,0 +1,1008 @@
+								if (terrOwnerPlacements == null)
+								{
+									terrOwnerPlacements = new LinkedHashMap<String, Integer>();
+									terrPlacements.put(owner, terrOwnerPlacements);
+								}
+								terrOwnerPlacements.put(attrUnitPlacements.get("unitType"), Integer.parseInt(attrUnitPlacements.get("quantity")));
+							}
+						}
+					}
+				}
+				stepToGo = Math.max(stepToGo, (s_unitPlacements.isEmpty() ? 13 : 14));
+			}
+			else if (childNodeName.equals("propertyList"))
+			{
+				final NodeList propertyListChildNodes = childNode.getChildNodes();
+				for (int prop_i = 0; prop_i < propertyListChildNodes.getLength(); ++prop_i)
+				{
+					final Node property = propertyListChildNodes.item(prop_i);
+					if (property.getNodeName().equals("property"))
+						parsePropertyNode(property);
+				}
+				if (!s_gameSettings.isEmpty())
+					stepToGo = (s_notes.length() > 0 ? 16 : 15);
+			}
+		}
+		return stepToGo;
+	}
+
+	private static void parsePropertyNode(final Node property)
+	{
+		final HashMap<String, String> propertyAttr = getAttributesMap(property.getAttributes());
+		final ArrayList<String> settingValues = new ArrayList<String>();
+		final String propertyName = propertyAttr.get("name");
+		if (propertyName.equals("notes") || propertyName.equals("mapName"))
+		{
+			final NodeList propertyListChildNodes = property.getChildNodes();
+			for (int prop_i = 0; prop_i < propertyListChildNodes.getLength(); ++prop_i)
+			{
+				final Node subProperty = propertyListChildNodes.item(prop_i);
+				if (subProperty.getNodeName().equals("value"))
+					s_notes = subProperty.getTextContent();
+			}
+			return;
+		}
+		settingValues.add(propertyAttr.get("value"));
+		settingValues.add(Boolean.toString(Boolean.parseBoolean(propertyAttr.get("editable"))));
+		final NodeList propertyNodes = property.getChildNodes();
+		for (int pr_i = 0; pr_i < propertyNodes.getLength(); ++pr_i)
+		{
+			final Node propertyRange = propertyNodes.item(pr_i);
+			if (propertyRange.getNodeName().equals("number"))
+			{
+				final HashMap<String, String> propertyRangeAttr = getAttributesMap(propertyRange.getAttributes());
+				settingValues.add(propertyRangeAttr.get("min"));
+				settingValues.add(propertyRangeAttr.get("max"));
+				s_gameSettings.put(propertyName, settingValues);
+				break;
+			}
+			else if (propertyRange.getNodeName().equals("boolean"))
+			{
+				settingValues.add("0"); // min
+				settingValues.add("0"); // max
+				s_gameSettings.put(propertyName, settingValues);
+				break;
+			}
+		}
+	}
+
+	private static void parseAttatchmentNode(final Node attatchment) {
+		final HashMap<String, String> attatchmentAttr = getAttributesMap(attatchment.getAttributes());
+		final String attatachmentName = attatchmentAttr.get("name");
+		final String attatachmentType = attatchmentAttr.get("type");
+		final String attatachmentAttatchTo = attatchmentAttr.get("attatchTo");
+		if (attatachmentName.equals("techAttatchment") && attatachmentType.equals("player"))

```

Comment:
Be warned, we have it in the pipeline to correct this incorrect spelling "attachment".